src/Ledger/Conway/Specification/Gov/Actions.lagda.md

```diff
@@ -441,3 +442,27 @@ proposedCC ⟦ UpdateCommittee , (x , _ , _) ⟧ᵍᵃ = dom x
 proposedCC _ = ∅
 ```
 
+```agda
+DReps : Type
+DReps = Credential ⇀ Epoch
+```
+
+<!--
+```agda
+record HasDReps {a} (A : Type a) : Type a where
+  field DRepsOf : A → DReps
+open HasDReps ⦃...⦄ public
+
+opaque
+```
+-->
+
+The following function takes a type `A`{.AgdaDatatype} with an associated set of
+`DReps`{.AgdaDatatype}, and an epoch `e`{.AgdaBound}, and returns the
+`DReps`{.AgdaDatatype} of `A`{.AgdaDatatype} that expire in epoch `e`{.AgdaBound} or
+later.
+
+```agda
+  DRepsOf_∣_≤Expiry : {A : Type} ⦃ _ : HasDReps A ⦄ → A → Epoch → DReps
+  DRepsOf a ∣ e ≤Expiry = filterᵐ (λ (_ , expEpoch) → e ≤ expEpoch) (DRepsOf a)
```

Comment:
I find confusing that the naming of this function uses a composite of names from other functions. By this I mean:
1. `DRepsOf`
2. `|`
3. `≤`